pkg/controller/snapshot_delete_test.go

```diff
@@ -165,8 +165,8 @@ func TestDeleteSync(t *testing.T) {
 		},
 		{
 			name:              "1-4 - fail delete with snapshot class that has invalid secret parameter",
-			initialContents:   newContentArray("content1-4", invalidSecretClass, "sid1-4", "vuid1-4", "volume1-4", "snapuid1-4", "snap1-4", &deletePolicy, nil, nil, true),
-			expectedContents:  newContentArray("content1-4", invalidSecretClass, "sid1-4", "vuid1-4", "volume1-4", "snapuid1-4", "snap1-4", &deletePolicy, nil, nil, true),
+			initialContents:   newContentArray("content1-4", invalidSecretClass, "sid1-4", "vuid1-4", "volume1-4", "snapuid1-4", "snap1-4", &deletePolicy, nil, nil, true, emptyNamespaceSecretAnnotations()),
+			expectedContents:  newContentArray("content1-4", invalidSecretClass, "sid1-4", "vuid1-4", "volume1-4", "snapuid1-4", "snap1-4", &deletePolicy, nil, nil, true, emptyNamespaceSecretAnnotations()),
```

Comment:
Is there a test case for if a secret is specified, but it doesn't exist?  To mirror the external-provisioner scenario, I believe we continue and pass no secret to the Delete call, and it's up to the plugin to fail if they need it.

Reply:
Unless I'm missing something, it seems that the current behavior for the snapshotter is to fail in this case:
https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/controller/snapshot_controller.go#L748-L751

Reply:
@ggriffiths I missed the provisioner behavior in this case.  Please go ahead to fix it and add a test case.  Thanks.

Reply:
Fixed and added deletion test `1-16` for continuing deletion. Also added a test for creation: `7-10`